sass/components/index-list.scss

```diff
@@ -0,0 +1,32 @@
+@import "mixins/index-list/index-list.scss";
+@import "mixins/index-list/index-list__item.scss";
+@import "mixins/index-list/index-list__item__index.scss";
+
+.index-list {
+  @include index-list;
+}
+
+.index-list__item {
+  @include index-list__item;
+}
+
+.index-list__item__index {
```

Comment:
I think what you are suggesting would only work if `index-list__item` was not a block, which sadly it is, therefore you get `index-list__item__index`